### DISCUSSION TITLE
Expose setValue via controllers

### DISCUSSION THREAD

**Discussion by dextermb** [react-hook-form/react-hook-form#7216] [Ideas]
Hi all

I'm currently working to get a date picker to conform to the standard of React Hook Form, but as you can imagine it's pretty finicky.

I'm curious why so many functions, state values and other stuff is exposed when using `render` from a `Controller` but not `setValue`. The Date picker I am using does not provide an `event` when `onChange` is fired, so unfortunately `field.onChange` would not work.

Thoughts?

Thanks

**Comment by bluebill1049** [comment]
provide some more context and example plz.

**Reply by dextermb** [reply]
You can find a general overview [here](https://gist.github.com/dextermb/d410d543f0fd9aa721265ec9987fb616). As you can see we have to [listen to an internal value change](https://gist.github.com/dextermb/d410d543f0fd9aa721265ec9987fb616#file-input-date-jsx-L82-L93) which then we use to fire a `onChange`, but as this is not a change event on an input we cannot easily trigger a change event.

So we end up using `Controller` and `useForm#setValue` within our [form](https://gist.github.com/dextermb/d410d543f0fd9aa721265ec9987fb616#file-new-request-jsx-L33-L45) but it would be nice if `Controller#render` exposed `setValue` rather than having to grab it from `useForm`.

Thoughts?

Edit: For some reason when linking to specific lines within a Gist it loses all indentation.
